pkg/audit/manager.go

```diff
@@ -371,30 +392,20 @@ func (am *Manager) auditResources(
 						continue
 					}
 
-					ns := corev1.Namespace{}
-					if objNamespace != "" {
-						ns, err = nsCache.Get(ctx, am.client, objNamespace)
-						if err != nil {
-							am.log.Error(err, "Unable to look up object namespace", "group", gv.Group, "version", gv.Version, "kind", kind)
-							continue
-						}
-					}
-
-					augmentedObj := target.AugmentedUnstructured{
-						Object:    objList.Items[index],
-						Namespace: &ns,
-					}
-					resp, err := am.opa.Review(ctx, augmentedObj)
+					// for each item, write object to a file along with the namespace
+					fileName := fmt.Sprintf("%s_%d", objNamespace, index)
```

Comment:
What purpose does embedding the namespace in the filename serve?

Reply:
From few lines above, since we are already calling the api server to get the namespace of the object to see if we can skip it to avoid having to create the file if namespace is excluded, we should persist the namespace as part of the filename so we do not have to make another api call later when loading the obj from disk. 
```
objNamespace := objList.Items[index].GetNamespace()
isExcludedNamespace, err := am.skipExcludedNamespace(&objList.Items[index])
```

Reply:
Two problems with this benefit:

1) We need to get the full namespace object so that we can evaluate namespace selectors, not the name
2) We are already caching the namespace in nsCache

Reply:
From few lines below, the namespace saved as part of the file name is used to lookup the full namespace object from nsCache:
```
objNs := strings.Split(fileName, "_")[0]
ns := corev1.Namespace{}
if objNs != "" {
  ns, err = nsCache.Get(ctx, am.client, objNs)
```

Reply:
I'm removing the namespace from the filename.